### ISSUE TITLE
Cupyx.profiler benchmark method to.str returns different inconsist formats.

### ISSUE THREAD

**Issue by Joachimoe** [cupy/cupy#7150]
### Description

Hi,

I am currently benchmarking cupy code and by reading the docs, I found that one should use the built-in ```cupyx.profiler``` benchmark. When calling benchmark on a Callable this returns an object of type ```_PerfCaseResult```. In order to post-process the results, I use the method ```to.str()```. However, this method inconsistently returns lists of different lengths which is problematic. 

### To Reproduce

import cupy as cp
import sys
from cupyx.profiler import benchmark
import cupyx.scipy.ndimage

global mat 
global arr 

def generate_matrix(size, work_units):
    rand = cp.random.default_rng()                            
    mat = rand.integers(0, 100_000, (size, size, work_units)) 
    return mat

def generate_array(size, work_units):
    rand = cp.random.default_rng()                            
    arr = rand.integers(0, 100_000, (size, work_units)) 
    return arr

def rotate(mat):
    y = cupyx.scipy.ndimage.rotate(mat, angle = 90) 
    sz = y.shape
    return 

def multiply(arr):
    y = cp.multiply(arr, 42)
    sz = len(y)
    return 
    

if __name__ == '__main__':
    sz = 100
    wk = 1_000
    mat = generate_matrix(sz, wk)  
    arr = generate_array(sz, wk)

    k = benchmark(rotate, (mat, ), n_repeat=20, n_warmup = 5)
    j = benchmark(rotate, (arr, ), n_repeat=20, n_warmup = 5)
    print(len(k.to_str(show_gpu=1).split())) #List of length 16
    print(len(j.to_str(show_gpu=1).split())) #List of length 24

### Environment

```
OS                           : Linux-5.15.0-52-generic-x86_64-with-glibc2.35
Python Version               : 3.9.13
CuPy Version                 : 10.6.0
CuPy Platform                : NVIDIA CUDA
NumPy Version                : 1.22.4
SciPy Version                : 1.9.1
Cython Build Version         : 0.29.30
Cython Runtime Version       : None
CUDA Root                    : /home/joachim/anaconda3/envs/rps
nvcc PATH                    : None
CUDA Build Version           : 11020
CUDA Driver Version          : 11070
CUDA Runtime Version         : 11050
cuBLAS Version               : (available)
cuFFT Version                : 10600
cuRAND Version               : 10207
cuSOLVER Version             : (11, 3, 2)
cuSPARSE Version             : (available)
NVRTC Version                : (11, 5)
Thrust Version               : 101000
CUB Build Version            : 101000
Jitify Build Version         : 87d8ccd
cuDNN Build Version          : None
cuDNN Version                : None
NCCL Build Version           : 21212
NCCL Runtime Version         : 21403
cuTENSOR Version             : None
cuSPARSELt Build Version     : None
Device 0 Name                : NVIDIA GeForce GTX 1660 SUPER
Device 0 Compute Capability  : 75
Device 0 PCI Bus ID          : 0000:01:00.0
```

**Comment by emcastillo** [comment]
Yeah, in some cases spaces are added while in others not, #7152 should fix this. Thanks for the report!

**Comment by emcastillo** [comment]
btw, you can access the cpu and GPU times directly from the object and print them in the format you want. This is to allow users to create easily parsable outputs, while the default `to_str` is mostly for being read by humans

**Comment by JoachimMoe** [comment]
@emcastillo When I try to access the objects ```gpu_times``` it looks like it returns the runs measured in MILLI seconds and not MICRO seconds which the object measures time in. Is this correct? When running the benchmark and printing the object, I get the following output:

Disclaimer: I set ```n_repeat = 20``` and ```n_warmup= 5```

This is the output when printing the entire object:
```
minmax              :    CPU:61295.709 us   +/-3834.641 (min:54187.836 / max:67313.261) us     GPU-0:61317.088 us   +/-3834.672 (min:54206.463 / max:67335.358) us
```

If I call ```.gpu_times``` on the object, the following is returned: 

```
[[0.06361789 0.06256678 0.06747869 0.06752739 0.07688797 0.07564902
  0.07257703 0.06614252 0.06904458 0.06727177 0.06762701 0.07100621
  0.06922637 0.07008909 0.06309142 0.07723645 0.0731935  0.07638221
  0.06922035 0.06834103]]
``` 

This is not written anywhere in any documentation, so I am not sure, but it very much looks like it is divided by 1000. This caught me very much off guard, as I divide by 1000 as documentation states it returns micro seconds. Should I open another issue addressing this?
